router.go

```diff
@@ -52,16 +52,19 @@ type MapRouter struct {
 	serviceProcedures         map[serviceProcedure]transport.Procedure
 	serviceProcedureEncodings map[serviceProcedureEncoding]transport.Procedure
 	supportedEncodings        map[serviceProcedure][]string
+	allServices               map[string]bool
 }
 
 // NewMapRouter builds a new MapRouter that uses the given name as the
 // default service name.
-func NewMapRouter(defaultService string) MapRouter {
+func NewMapRouter(defaultService string) (m MapRouter) {
+	defer func() { m.allServices[defaultService] = true }()
 	return MapRouter{
 		defaultService:            defaultService,
 		serviceProcedures:         make(map[serviceProcedure]transport.Procedure),
 		serviceProcedureEncodings: make(map[serviceProcedureEncoding]transport.Procedure),
 		supportedEncodings:        make(map[serviceProcedure][]string),
+		allServices:               make(map[string]bool),
```

Comment:
Since we just want to know if a service exists, let's change this type to a `map[string]struct{}`. This actually uses less space since `struct{}` uses no space. See: https://dave.cheney.net/2014/03/25/the-empty-struct

Reply:
Changed map definition. Thanks for the suggestion.